src/main/java/org/elasticsearch/index/analysis/SkroutzGreekStemmer.java

```diff
@@ -64,55 +62,28 @@
  * about these suffixes are added.
  */
 public class SkroutzGreekStemmer {
-  public final static String DEFAULT_STOPWORD_FILE = "stopwords.txt";
-  private final CharArraySet stopwords;
-  private Map<String, char[]> stepZeroExceptions;
+  private CharArraySet protectedWords = CharArraySet.EMPTY_SET;
+  private CharArrayMap stepZeroExceptions = CharArrayMap.emptyMap();
 
-  public SkroutzGreekStemmer(final CharArraySet stopwords) {
-    this.stopwords = stopwords;
+  public SkroutzGreekStemmer() {
     try {
-      this.stepZeroExceptions = loadExceptions("step_0_exceptions");
+      this.protectedWords = new CharArraySet(
+              loadProtected("protected_words"), true);
+      this.stepZeroExceptions = new CharArrayMap(
+              loadExceptions("step_0_exceptions"), true);
+
     } catch(Exception ex) {
       System.err.println("Failed to load exceptions from yaml file");
-      ex.printStackTrace();
-      this.stepZeroExceptions =  new HashMap(0);
-    }
-  }
-
-  public SkroutzGreekStemmer() {
-    this(SkroutzGreekStemmer.getDefaultStopSet());
-  }
-
-  public static final CharArraySet getDefaultStopSet() {
-    return DefaultSetHolder.DEFAULT_SET;
-  }
-
-  private static class DefaultSetHolder {
-    private static final CharArraySet DEFAULT_SET;
-
-    static {
-      try {
-        DEFAULT_SET = loadStopwordSet(
-            SkroutzGreekStemmer.class.getResourceAsStream(DEFAULT_STOPWORD_FILE),
-            Lucene.VERSION);
-      } catch (IOException ex) {
-        // default set should always be present as it is part of the
-        // distribution (JAR)
-        throw new RuntimeException("Unable to load default stopword set");
-      }
     }
   }
 
   public int stem(char s[], int len) {
-    // Too short or a stopword
-    if (len < 3 || stopwords.contains(s, 0, len))
+    // Too short or a protected word
+    if (len < 3 || protectedWords.contains(s, 0, len))
       return len;
 
-    // a String representation of the term's char[]
-    String termStr = String.valueOf(s, 0, len);
-
     //handle step 0 and step 1 exceptions
-    char[] exceptional = stepZeroExceptions.get(termStr);
+    char[] exceptional = (char[]) stepZeroExceptions.get(s, 0, len);
```

Comment:
Do you need to cast here? I thought `stepZeroExceptions` values already are `char` arrays.